stdlib/SparseArrays/src/higherorderfns.jl

```diff
@@ -1154,4 +1155,23 @@ map(f::Tf, A::SparseOrStructuredMatrix, Bs::Vararg{SparseOrStructuredMatrix,N})
 map!(f::Tf, C::AbstractSparseMatrixCSC, A::SparseOrStructuredMatrix, Bs::Vararg{SparseOrStructuredMatrix,N}) where {Tf,N} =
     (_checksameshape(C, A, Bs...); _noshapecheck_map!(f, C, _sparsifystructured(A), map(_sparsifystructured, Bs)...))
 
+
+# (13) extrema methods optimized for sparse vectors/matrices.
+function extrema(f, A::SparseVecOrMat)
```

Comment:
This doesn't handle `dims` or `dims=:`, so I believe those will still call the slower generic code. We might want to at least handle `dims=:`, and use `invoke` to call the AbstractArray method for general dims.

Reply:
I think it's worse because `dims` is a keyword argument, so this will error now if you pass a `dims` argument. This will probably need to be renamed to a hidden function `_extrema` and `extrema` needs to check for a `dims` argument and fall back to the slow code if `dims` is not `:`. An alternative of course would be to handle the `dims` argument here directly.

Reply:
It doesn't throw errors. It only use the slow generic code.
There are only thew cases.
`extrema(f, A::SparseVector, dims=1)`, `extrema(f, A::AbstractSparseMatrix, dims=[1,2])` and `extrema(f, SparseVecOrMat, ::Colon)`
which are equivalent to `extrema(f, A::SparseVecOrMat)`.
We still need to implement `extrema(f, A::SparseVecOrMat, dims=1)` and `extrema(f, A::SparseVecOrMat, dims=2)`.
In my opinion there should be two separate functions because the optimized algorithms are quite different.
I believe it covers every cases.

Reply:
The problem is that `dims` is not a positional argument and keyword arts don't participate in dispatch. The correct function signature should probably be `Base._extrema_dims(f, A::SparseVecOrMat, ::Colon)`, so this only gets called for the `dims=:` case (which is also the default).

Reply:
I implemented the @JeffBezanson's idea of using `invoke`. Actually reimplementing the functions that handles dims would greatly improve speed but it is a bit tricky to get them consistent with the Base implementation.